### PR TITLE
Update release process for dot release

### PR DESCRIPTION
**Pull request by jasonblais**
Add a checklist item for marketing team to prepare blog post, and email/Twitter announces for each dot release

Also corrects a grammar inconsistency

### CODE REVIEW

File: source/process/release-process.md [hunk 1]
```diff
@@ -249,8 +249,11 @@ Once final dot release build is ready to cut:
     - Update the version archive in the [upgrade guide](https://github.com/mattermost/docs/blob/master/source/administration/upgrade.md)
     - Sumbit GitLab MR to update the version number and MD5 hash to the dot release version. [See example](https://gitlab.com/gitlab-org/omnibus-gitlab/merge_requests/1127)
       - [Test the upgrade](https://docs.google.com/document/d/1mbeu2XXwCpbz3qz7y_6yDIYBToyY2nW0NFZq9Gdei1E/edit#heading=h.ncq9ltn04isg) once the MR is merged and the package is released to the GitLab package server
-3.  Logistics:  
-    - Verifies each of the issues in the patch release are fixed
+3. Logistics:  
+    - Verify each of the issues in the patch release are fixed
+4. Marketing:
+    - Prepare blog post for mattermost.com and set timer for 08:00 PST on the day after dot release
+    - Prepare MailChimp email blast and Twitter announcement and send for marketing lead to review. Once reviewed, schedule for 08:00 PST on the day after dot release
```
Comment (it33) [inline source/process/release-process.md]: Maybe add that if there is a security update in the dot release to mention it an include a recommendation to update.

Comment (jasonblais) [inline source/process/release-process.md]: Thanks! Agree

File: source/process/release-process.md [hunk 2]
```diff
@@ -249,8 +249,11 @@ Once final dot release build is ready to cut:
     - Update the version archive in the [upgrade guide](https://github.com/mattermost/docs/blob/master/source/administration/upgrade.md)
     - Sumbit GitLab MR to update the version number and MD5 hash to the dot release version. [See example](https://gitlab.com/gitlab-org/omnibus-gitlab/merge_requests/1127)
```
Comment (JeffSchering) [inline source/process/release-process.md]: Typo here... *Sumbit* should be *Submit*

Comment (jasonblais) [inline source/process/release-process.md]: Nice catch :) Fixed

**Comment by lfbrock** [comment]
+1

**Comment by esethna** [comment]
+1

**Pull request merged**